test/unit/config/test__base.py

```diff
@@ -204,50 +205,53 @@ def test_prepare_env_ansible_vars(mocker):
         "PYTHONPATH is the union of the explicit env['PYTHONPATH'] override and AWX_LIB_DIRECTORY"
 
 
-def test_prepare_with_ssh_key(mocker):
+def test_prepare_with_ssh_key(mocker, tmp_path):
     open_fifo_write_mock = mocker.patch('ansible_runner.config._base.open_fifo_write')
-    mocker.patch.dict('os.environ', {'AWX_LIB_DIRECTORY': '/tmp/artifact'})
+    custom_artifacts = tmp_path.joinpath('custom_arts')
 
-    rc = BaseConfig(private_data_dir='/tmp')
-    rc.artifact_dir = '/tmp/artifact'
+    rc = BaseConfig(private_data_dir=tmp_path.as_posix(), artifact_dir=custom_artifacts.as_posix())
+    rc.artifact_dir = custom_artifacts.as_posix()
     rc.env = {}
     rc.execution_mode = BaseExecutionMode.ANSIBLE_COMMANDS
     rc.ssh_key_data = '01234567890'
     rc.command = 'ansible-playbook'
     rc.cmdline_args = []
     rc._prepare_env()
 
-    assert rc.ssh_key_path == '/tmp/artifact/ssh_key_data'
+    assert rc.ssh_key_path == custom_artifacts.joinpath('ssh_key_data').as_posix()
     assert open_fifo_write_mock.called
 
 
-def test_wrap_args_with_ssh_agent_defaults():
-    rc = BaseConfig(private_data_dir='/tmp')
-    res = rc.wrap_args_with_ssh_agent(['ansible-playbook', 'main.yaml'], '/tmp/sshkey')
+def test_wrap_args_with_ssh_agent_defaults(tmp_path):
+    tp = tmp_path.as_posix()
```

Comment:
This doesn't need to be converted to a string prematurely. The `__str__` method is called when a `Path` object used in an `f` string. 
```suggestion
    tp = tmp_path
```